src/spikeinterface/postprocessing/template_metrics.py

```diff
@@ -139,12 +144,34 @@ def _set_params(
             metrics_kwargs_ = _default_function_kwargs.copy()
             metrics_kwargs_.update(metrics_kwargs)
 
+        metrics_to_compute = metric_names
+        tm_extension = self.sorting_analyzer.get_extension("template_metrics")
+        if delete_existing_metrics is False and tm_extension is not None:
+
+            existing_params = tm_extension.params["metrics_kwargs"]
+            # checks that existing metrics were calculated using the same params
+            if existing_params != metrics_kwargs_:
+                warnings.warn(
```

Comment:
In defence of the warning:
Suppose a user is going through a large set of `sorting_analyzer`s that they computed over a long time span. They want to compute a new `template_metric` that they’ve heard about. They know it takes a long time to recompute everything, so they just specify the new metric to compute. Sometimes all the other metrics are deleted! Sometimes they are not!! What is going on?? This warning would have told them. Plus it’s quite rare to get the warning. The default behaviour (compute everything) will never trigger it.

Another option: if the kwargs are different, there is an error telling the user to either use the same kwargs are before or to include all the previous metrics in their compute (quite a complicated error)

Reply:
I am not familiar enough with the sorting analyzer workflow to have a useful opinion on this one.  

I think it would be better if you had the parameters (the new and the old) in the argument to give the user a better workflow.

Something to think about as well is whether this will be called in a context where the warning will make no sense at all for the user. Say, somebody wraps a sorting analyzer and keeps recomputing then the user gets a warning that they don't know anything about. For those cases, it is good to have an option to turn off a sure warning. 

But I don't really know how this is used to judge if this last concern is valid or not.

Reply:
Ooh yes, I like the idea of printing the old and new parameters!